web/packages/teleterm/src/ui/Vnet/VnetConnectionItem.tsx

```diff
@@ -0,0 +1,235 @@
+            bold
+            color="text.main"
+            css={`
+              line-height: 16px;
+            `}
+          >
+            <span
+              css={`
+                vertical-align: middle;
+              `}
+            >
+              VNet
+            </span>
+          </Text>
+          <Text color="text.slightlyMuted" typography="body2">
+            Virtual Network Emulation
+          </Text>
+        </div>
+
+        {/* Buttons to the right. Negative margin to match buttons of other connections. */}
+        <Flex gap={1} mr="-3px">
+          {props.showHelpButton && (
+            <ButtonIcon
+              as="a"
+              title="Open VNet documentation"
+              href="https://goteleport.com/docs/connect-your-client/teleport-connect/#vnet"
+              target="_blank"
+              onClick={e => {
+                // Don't trigger ListItem's onClick.
+                e.stopPropagation();
+              }}
+            >
+              <icons.Question size={18} />
+            </ButtonIcon>
+          )}
+
+          {startAttempt.status === 'processing' ||
+          stopAttempt.status === 'processing' ? (
+            <icons.Spinner
+              css={`
+                width: 32px;
+                height: 32px;
+                animation: ${rotate360} 1.5s infinite linear;
+              `}
+              size={18}
+            />
+          ) : (
+            <>
+              {status === 'running' && (
+                <ButtonIcon
+                  title="Stop VNet"
+                  onClick={e => {
+                    e.stopPropagation();
+                    stop();
+                  }}
+                >
+                  <icons.BroadcastSlash size={18} />
+                </ButtonIcon>
+              )}
+              {status === 'stopped' && (
+                <ButtonIcon
+                  title="Start VNet"
+                  onClick={e => {
+                    e.stopPropagation();
+                    start();
+                  }}
+                >
+                  <icons.Broadcast size={18} />
+                </ButtonIcon>
+              )}
+            </>
+          )}
+        </Flex>
+      </Flex>
+    </ListItem>
+  );
+};
+
+/**
+ * AppConnectionItem is an individual connection to an app made through VNet, shown in
+ * VnetSliderStep.
+ */
+export const AppConnectionItem = (props: {
+  app: string;
+  status: 'on' | 'error' | 'off';
+  error?: string;
+}) => {
+  const { notificationsService } = useAppContext();
+
+  const copy = async () => {
+    const content = [props.app, props.error].filter(Boolean).join(': ');
+    await copyToClipboard(content);
+
+    notificationsService.notifyInfo(
+      props.error
+        ? `Copied error for ${props.app} to clipboard`
+        : `Copied ${props.app} to clipboard`
+    );
+  };
+
+  return (
+    <ListItem
+      css={`
+        padding: 6px 8px;
+        height: unset;
+      `}
+      onClick={copy}
```

Comment:
When I saw it for the first time, I was a bit surprised that clicking on the item copied it to the clipboard, instead of opening "something". I think a better UX would be to just have a copy button on hover.

Reply:
There's nothing to open though, the most common use case around an app address when using VNet is copying it and then pasting it outside of Connect.

But I do agree that it'd be better to have a dedicated button for that. It's seems better than informing the user through `title` that it's how they can copy the address.

I'll add a copy button that gets shown on hover.